dotCMS/src/main/java/com/dotmarketing/factories/MultiTreeFactory.java

```diff
@@ -25,691 +16,343 @@
  * </p>
  * 
  * @author will
  */
 public class MultiTreeFactory {
-    
-    private static final String DELETE_MULTITREE_ERROR_MSG = "Deleting MultiTree Object failed:";
-    private static final String SAVE_MULTITREE_ERROR_MSG = "Saving MultiTree Object failed:";
 
-	public static void deleteMultiTree(Object o1, Object o2, Object o3) {
-		Inode inode1 = (Inode) o1;
-		Inode inode2 = (Inode) o2;
-		Inode inode3 = (Inode) o3;
+    final static String DELETE_SQL = "delete from multi_tree where parent1=? and parent2=? and child=? and  relation_type = ?";
+    final static String SELECT_SQL =
+            "select * from multi_tree where parent1 = ? and parent2 = ? and child = ? and  relation_type = ?";
 
-		try {
+    final static String INSERT_SQL =
+            "insert into multi_tree (parent1, parent2, child, relation_type, tree_order ) values (?,?,?,?,?)  ";
 
+    final static String SELECT_BY_ONE_PARENT = "select * from multi_tree where parent1 = ? or parent2 = ? ";
+    final static String SELECT_BY_TWO_PARENTS = "select * from multi_tree where parent1 = ? and parent2 = ?  order by tree_order";
+    final static String SELECT_ALL = "select * from multi_tree  ";
+    final static String SELECT_BY_CHILD = "select * from multi_tree where child = ?  order by parent1, parent2, relation_type ";
+    final static String SELECT_BY_PARENTS_AND_RELATIONS =
+            " select * from multi_tree where parent1 = ? and parent2 = ? and relation_type = ? order by tree_order";
 
-			DotConnect db = new DotConnect();
-			db.setSQL("delete from multi_tree where parent1 =? and parent2 = ? and child = ? ");
-			db.addParam(inode1.getInode());
-			db.addParam(inode2.getInode());
-			db.addParam(inode3.getInode());
-			db.getResult();
 
-			updateHTMLPageVersionTS(inode1.getInode());
-			refreshPageInCache(inode1.getInode());
-			
-		}
-		catch (Exception e) {
-			throw new DotRuntimeException(e.getMessage());
-		}
-
-	}
+    public static void deleteMultiTree(final MultiTree mTree) throws DotDataException {
+        _dbDelete(mTree);
+        updateHTMLPageVersionTS(mTree.getHtmlPage());
+        refreshPageInCache(mTree.getHtmlPage());
+    }
 
     /**
-     * Just invoking deleteMultiTreeByParent1 with null language.
-     * @throws DotSecurityException 
+     * Deletes multi-tree relationship given a MultiTree object. It also updates the version_ts of
+     * all versions of the htmlpage passed in (multiTree.parent1)
+     *
+     * @param multiTree
+     * @throws DotDataException
+     * @throws DotSecurityException
      * 
-     * @see MultiTreeFactory#deleteMultiTreeByParent1(Identifier parent, Long
-     *      languageId)
      */
-    public static void deleteMultiTreeByParent1(Contentlet contentlet) throws DotDataException, DotSecurityException {
-        Identifier identifier = new Identifier();
-        identifier.setId(contentlet.getIdentifier());
+    private static void _dbDelete(final MultiTree mTree) throws DotDataException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00100)